docs/data/data-grid/server-side-data/tree-data.md

```diff
@@ -2,14 +2,36 @@
 title: React Server-side tree data
 ---
 
-# Data Grid - Server-side tree data [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')🚧
+# Data Grid - Server-side tree data [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')
 
 <p class="description">Tree data lazy-loading with server-side data source.</p>
 
-:::warning
-This feature isn't implemented yet. It's coming.
+To use the server-side tree data, pass the `unstable_dataSource` prop as explained in the [overview section](/x/react-data-grid/server-side-data/), in addition to that passing of some additional props is required for the server-side tree data to work properly.
 
-👍 Upvote [issue #3377](https://github.com/mui/mui-x/issues/3377) if you want to see it land faster.
+- `getGroupKey(row: GridRowModel): string`
 
-Don't hesitate to leave a comment on the same issue to influence what gets built. Especially if you already have a use case for this component, or if you are facing a pain point with the [currently proposed workaround](https://mui.com/x/react-data-grid/tree-data/#children-lazy-loading).
+  Used by the grid to group rows by their parent group. Replaces `getTreeDataPath` used in client-side tree-data.
+  For example, consider this tree structure for tree data.
+
+  ```js
+  - (1) Sarah // groupKey 'Sarah'
+    - (2) Thomas // groupKey 'Thomas'
+  ```
+
+  When `(2) Thomas` is expanded, the `getRows` function will be called with group keys `['Sarah', 'Thomas']`.
+
+- `hasChildren(row: GridRowModel): boolean`
+
+  Used by the grid to check if a row has children on server
```

Comment:
Would this be clearer since it's meant to use a key from the row, not querying on the server?
```suggestion
  Used to check if a row has children. It receives `row` as a parameter.
```